doc/decisions/DR016-One-to-many-result-paging.md

```diff
@@ -0,0 +1,501 @@
+        batch: List[EntityReference],
+        relationship,
+        successCb: SuccessCallback,
+        errorCb,
+        limit: int = -1, offset: int = 0): ...
+```
+
+##### Pros
+
+- Conceptually simple for the host.
+- Jumping arbitrarily to nth page is trivial.
+- Retains batch-awareness for each page.
+- Transactions, indexing and caching technologies exist that can
+  alleviate stability and performance problems, to some extent.
+
+##### Cons
+
+- Anomalous page results if insertions/deletions can happen before next
+  page is retrieved, or if ordering is not stable.
+- Server cannot negotiate page size - there is no mechanism to signal
+  that more results are available other than by exhausting the page
+  limit.
+- Poor performance on the back-end for large offsets into non-trivial
+  (i.e. non-indexed/cached) queries, where many rows must be iterated
+  through and skipped over.
+
+## Options considered
+
+### Option 1 - Page token
+
+Firstly let's consider the general pros and cons of page tokens.
+
+#### Pros
+
+- Conceptually simple.
+- Matches the most common pattern of pageable HTTP APIs allowing e.g.
+  the page token to be a thin wrapper around a "real" page token(s).
+- Manager plugin can remain conceptually stateless, since required state
+  is bundled up in the page token.
+- Page token is (or can be) immutable, aiding thread-safety.
+- Using the page token to determine if more pages are available (e.g.
+  a `None` token means exhausted) allows the server to override the
+  page size, opening up an avenue for optimisation.
+
+#### Cons
+
+- Adding more functionality (e.g. previous page) requires additional
+  sibling functions or arguments.
+- Adding metadata (e.g. total results) requires additional sibling
+  functions or arguments.
+- Function invocations to fetch subsequent pages must pass all the same
+  arguments, such that they are valid for the page token. This is
+  error-prone and/or redundant.
+
+
+#### Option 1.1 - Whole-batch page token
+
+Host supplies a batch of entity references and a callback. The callback
+is called for each input entity, with one of its arguments being a
+page of related entities as a simple list.
+
+Once callbacks have been called, the function returns a page token. This
+can be supplied to a subsequent call to get the next page of results for
+all entities in the input batch simultaneously.
+
+Note that this could also be designed to use a cursor object with
+similar pros and cons, but that option was discarded (see Background
+section).
+
+```python
+class AbstractPageToken:
+    ...
+
+# Callback taking index of batch element and list of related entities.
+SuccessCallback = Callable[[int, List[EntityReferences]], None]
+
+def relatedEntities(
+        batch: List[EntityReference],
+        relationship,
+        successCb: SuccessCallback,
+        errorCb,
+        limit: int = -1,
+        pageToken: AbstractPageToken = None) -> AbstractPageToken: ...
+```
+
+```python
+# Usage:
```

Comment:
Added usage examples (here and for other options) after offline conversation in e9e343ab

Reply:
Don't worry about spending too much time on the batch, paged illustrations,  as we know were going to exclude them for now, def need to document manager implementation sketches for token vs cursor in singular form.